tensorflow/core/framework/types.h

```diff
@@ -68,8 +68,9 @@ class DeviceType {
 std::ostream& operator<<(std::ostream& os, const DeviceType& d);
 
 // Convenient constants that can be passed to a DeviceType constructor
-extern const char* const DEVICE_CPU;  // "CPU"
-extern const char* const DEVICE_GPU;  // "GPU"
+extern const char* const DEVICE_CPU;   // "CPU"
+extern const char* const DEVICE_GPU;   // "GPU"
+extern const char* const DEVICE_SYCL;  // "SYCL"
```

Comment:
I guess since we're registering all the kernels in the same files as CPU and GPU, this is fine.  Otherwise I'd have suggested defining something like a sycl/defs.h header where common SYCL-specific defines could go.